### PR TITLE
New bar plot API

### PR DESCRIPTION
**Pull request by mrkn**
I'd like to add the new plotting APIs.  This pull-request consists of the following things.

- `Charty.use_backend` method and `Charty::Backends.current` property
  - These allow us to manage "the current backend"
- Tentative new backend API
  - This is performed on only Pyplot backend so that I can implement new bar plot API
- `Charty.bar_plot` method and `Charty::Plotters::BarPlotter` class

### CODE REVIEW

**Comment by mrkn** [comment]
Now I've done the first-step implementation of `Charty.bar_plot` method.

The usage example is available in this gist:
https://gist.github.com/mrkn/2438b51965d5f17ae7a8e8f84d9a74a2

File: test/plot_methods/bar_plot_test.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      y = [10, 20, 30, 25, 15]
+      fig = Charty.bar_plot(x, y)
+      assert_instance_of(Charty::Plotters::BarPlotter, fig)
+      assert_equal(x, fig.x)
+      assert_equal(y, fig.y)
+      assert_equal(nil, fig.color)
+      assert_equal(nil, fig.data)
+      assert_equal(nil, fig.palette)
+      assert_equal(x, fig.group_names)
+      assert_equal(y.map{|v| [v] }, fig.plot_data)
```
Comment (kou) [inline test/plot_methods/bar_plot_test.rb]: One `assert_equal` is better for easy to debug:

```ruby
assert_equal({
               x: x,
               y: y,
               color: nil,
               data: nil,
               palette: nil,
               group_names: x,
               plot_data: y.map{|v| [v] },
             },
             {
               x: fix.x,
               y: fix.y,
               color: fig.color,
               data: fig.data,
               palette: fig.palette,
               group_names: fig.group_names,
               plot_data: fig.plot_data,
             })
```

Comment (mrkn) [reply test/plot_methods/bar_plot_test.rb]: Sounds reasonable.

File: lib/charty/backend_methods.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+module Charty
+  module BackendMethods
+    def use_backend(backend)
```
[the remark this answers is not in the document]
Comment (mrkn) [reply lib/charty/backend_methods.rb]: OK.

**Pull request merged**